gradio/components.py

```diff
@@ -2082,49 +2098,152 @@ def preprocess(self, x: Dict[str, str] | None) -> str | None:
         else:
             return str(file_name)
 
-    def postprocess(self, y: str | None) -> Dict[str, Any] | None:
+    def postprocess(
+        self, y: str | Tuple[str, str | None] | None
+    ) -> Tuple[FileData | None, FileData | None] | None:
         """
         Processes a video to ensure that it is in the correct format before
         returning it to the front end.
         Parameters:
-            y: a path or URL to the video file
+            y: video data in either of the following formats: a tuple of (str video filepath, str subtitle filepath), or a string filepath or URL to an video file, or None.
         Returns:
-            a dictionary with the following keys: 'name' (containing the file path
-            to a temporary copy of the video), 'data' (None), and 'is_file` (True).
+            a tuple with the two dictionary, reresent to video and (optional) subtitle, which following formats:
+            - The first dictionary represents the video file and contains the following keys:
+                - 'name': a file path to a temporary copy of the processed video.
+                - 'data': None
+                - 'is_file': True
+            - The second dictionary represents the subtitle file and contains the following keys:
+                - 'name': None
+                - 'data': Base64 encode the processed subtitle data.
+                - 'is_file': False
+            - If subtitle is None, returns (video, None).
+            - If both video and subtitle are None, returns None.
+        """
+
+        if y is None or y == [None, None] or y == (None, None):
+            return None
+        if isinstance(y, str):  # type: string
```

Comment:
```suggestion
        if isinstance(y, str):
```
(unnecessary comment)